lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,256 @@
+    required: false
+    default: ''
+  remote:
+    description:
+      - "The flatpak remote repo to be used in the flatpak operation."
+    required: false
+    default: ''
+  state:
+    description:
+      - "Set to `present` will install the flatpak and/or remote.
+        Set to `absent` will remove the flatpak and/or remote."
+    required: false
+    default: present
+'''
+EXAMPLES = '''
+
+#install the spotify flatpak
+
+ - flatpak:
+    name:  https://s3.amazonaws.com/alexlarsson/spotify-repo/spotify.flatpakref
+    state: present
+
+#add the gnome remote andd install gedit flatpak
+ - flatpak:
+    name: https://git.gnome.org/browse/gnome-apps-nightly/plain/gedit.flatpakref
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+#remove the gedit flatpak and remote
+ - flatpak:
+    name: org.gnome.gedit
+    remote: https://sdk.gnome.org/gnome-apps.flatpakrepo
+    state: absent
+
+#remove the gedit package
+ - flatpak:
+    name: org.gnome.gedit
+    state: absent
+
+'''
+RETURN = '''
+reason:
+    description: on failure, the output for the failure
+    returned: failed
+    type: string
+    sample: "error while installing..."
+name:
+    description: "Name of flatpak given for the operation"
+    returned: always
+    type: string
+    sample: "https://git.gnome.org/.../gnome-apps/gedit.flatpakref"
+remote:
+    description: "remote of flatpak given for the operation"
+    returned: always
+    type: string
+    sample: "https://sdk.gnome.org/gnome-apps.flatpakrepo"
+'''
+import subprocess
+from ansible.module_utils.basic import AnsibleModule
+
+
+def install_flat(binary, flat):
+    command = "{} install -y --from {}".format(binary, flat)
+    output = flatpak_command(command)
+    if 'error' in output and 'already installed' not in output:
+        return 1, output
+    return 0, output
+
+
+def uninstall_flat(binary, flat):
+    # This is a difficult function because it seems there
+    # is no naming convention for the flatpakref to what
+    # the installed flatpak will be named.
+    common_name = parse_flat(flat)
+    command = "{} list --app".format(binary)
+    process = subprocess.Popen(
+        command.split(), stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    output = subprocess.check_output(
+        ("grep", "-i", common_name), stdin=process.stdout)
+    process.wait()
+    installed_flat_name = output.split('/')[0]
+    command = "{} uninstall {}".format(binary, installed_flat_name)
+    output = flatpak_command(command)
+    if 'error' in output and 'not installed' not in output:
+        return 1, output
+    return 0, output
+
+
+def parse_remote(remote):
+    name = remote.split('/')[-1]
+    if '.' not in name:
+        return name
+    return name.split('.')[0]
+
+
+def parse_flat(name):
+    if 'http://' in name or 'https://' in name:
+        app = name.split('/')[-1].split('.')[0]
+        common_name = app
+    else:
+        common_name = name
+
```

Comment:
Either remove this empty line OR add the same empty line into all function in front of the return command (preferred solution).